sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -395,12 +434,11 @@ object GeneratorNestedColumnAliasing {
             val updatedGenerate = rewrittenG.copy(generatorOutput = updatedGeneratorOutput)
 
             // Replace nested column accessor with generator output.
-            val attrExprIdsOnGenerator = attrToExtractValuesOnGenerator.keys.map(_.exprId).toSet
+            val generateOutputAttrExprId = attrToExtractValuesOnGenerator.keys.map(_.exprId).head
```

Comment:
I changed the toSet and find way to toSet.head to make it clearer - we should only have one generator output here.